reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Zip.kt

```diff
@@ -30,7 +27,7 @@ fun <T, R> Iterable<Observable<T>>.zip(mapper: (List<T>) -> R): Observable<R> =
         emitter.setDisposable(disposables)
 
         val values = List<ArrayList<T>>(sources.size) { ArrayList() }
-        val completed = AtomicList(List(sources.size) { false })
+        val completed = MutableList(sources.size) { false }
```

Comment:
Also probably makes sense to revisit operators that works with ArrayLists of fixed size and replace with arrays. Almost free win at least on boxing-unboxing.

But up to you.

Reply:
Yes, this makes sense. Will update this one.